authentik/policies/unique_password/signals.py

```diff
@@ -0,0 +1,49 @@
+"""authentik policy signals"""
+
+from typing import Any
+
+from django.db.models.signals import post_delete
+from django.dispatch import receiver
+from django.http import HttpRequest
+
+from authentik.core.middleware import SESSION_KEY_IMPERSONATE_USER
+from authentik.core.models import User
+from authentik.policies.models import PolicyBinding
+from authentik.policies.unique_password.tasks import (
+    purge_password_history_table,
+    trim_user_password_history,
+)
+from authentik.stages.user_write.signals import user_write
+
+
+@receiver(post_delete, sender=PolicyBinding)
+def purge_password_history(sender, instance, **_):
+    from authentik.policies.unique_password.models import UniquePasswordPolicy
+
+    if not isinstance(instance.policy, UniquePasswordPolicy):
+        return
+    purge_password_history_table.delay()
+
+
+@receiver(user_write)
```

Comment:
There is a `password_changed` signal you can listen on. It's sent from `authentik/core/models.py::User::set_password`

Reply:
It will also allow you to ignore password changes in case `request.user != user`, in which case the user's password was changed by an admin from the admin interface, and should probably (?) be ignored